src/ItemList.js

```diff
@@ -61,11 +118,38 @@ function ItemList(props) {
           )}
           <h2>Shopping List:</h2>
           <form>
-            <ul>
-              {props.list.map((item) => (
-                <Item key={item.id} userToken={props.userToken} item={item} />
-              ))}
-            </ul>
+            {queryObj['week'].length > 0 && (
+              <ul>
+                <h3>Items to buy in the next week:</h3>
+                {queryObj['week'].map((item) => (
+                  <Item key={item.id} userToken={props.userToken} item={item} />
+                ))}
+              </ul>
+            )}
+            {queryObj['month'].length > 0 && (
+              <ul>
+                <h3>Items to buy in the next month:</h3>
+                {queryObj['month'].map((item) => (
+                  <Item key={item.id} userToken={props.userToken} item={item} />
+                ))}
+              </ul>
+            )}
+            {queryObj['longer'].length > 0 && (
+              <ul>
+                <h3>Items to buy in the distant future:</h3>
+                {queryObj['longer'].map((item) => (
+                  <Item key={item.id} userToken={props.userToken} item={item} />
+                ))}
+              </ul>
+            )}
+            {queryObj['inactive'].length > 0 && (
+              <ul>
+                <h3>Purchases we can't predict yet:</h3>
+                {queryObj['inactive'].map((item) => (
+                  <Item key={item.id} userToken={props.userToken} item={item} />
+                ))}
+              </ul>
```

Comment:
Again, just a comment, not a request. There is a lot of code duplication here ^, which _does_ make it easy to read. I wonder if you could use `Object.entries` on the `queryObj` and then map through that. Maybe too fancy for this. I like how this reads.